### DISCUSSION TITLE
Add "image_url" to Langfuse Prompt types / multi-modal / content-parts

### DISCUSSION THREAD

**Discussion by grobruegge** [langfuse/langfuse#3074] [Ideas]
### Describe the feature or potential improvement

Prompts within Langfuse currently support the types "system", "user" and "assistant". While this is sufficient within the context of pure language generation, modern "language" models such as "GPT-4o" are multi-model and also support images. Within the Python SDK, images can be passed to the model using `type=image_url` within messages as role *user*.

Langfuse does not natively support image urls within their prompt management, which requires some boilerplate code. For example, adding an image url in Python will look as follows:

```Python
compiled_prompt = langfuse_prompt.compile(
    attribute1=somevalue, attribute2=someothervalue
)

response = openai_client.chat.completions.create(
    model="gpt-4o",
    messages=[
        {
            "role": "system",
            "content": compiled_prompt[0]["content"],
        },
        {
            "role": "user",
            "content": [
                {
                    "type": "text",
                    "text": compiled_prompt[1]["content"],  # user prompt
                },
                {
                    "type": "image_url",
                    "image_url": {
                        "url": some_url,
                        "detail": "low",
                    },
                },
            ],
        },
    ],
)
```

This adds quiet a lot of code. It would be nice if one could simple add the image_url during the compile step which would then look as follows (only a suggestion):

```Python
compiled_prompt = langfuse_prompt.compile(
    attribute1=somevalue, attribute2=someothervalue, img_url=someurl
)
response = openai_client.chat.completions.create(
    model="gpt-4o",
    messages=compiled_prompt
)
```

Thank you so much for this framework and I am exited to see this in future versions of Langfuse!

**Comment by marliessophie** [comment]
Hi @grobruegge, thanks so much for providing this feedback. 

Agree, the DX here should be improved. We'll look into this in the next few weeks to provide a simpler solution. For now, your implementation is the way to go. We'll keep you posted here when we make the corresponding changes 👍

**Comment by squarebridges** [comment]
Just following up on this as I've come across a similar use case.

Any plans to support media (images, PDFs, anything that can be base64 encoded) as part of prompts?

**Reply by marcklingen** [reply]
on the roadmap, thanks for adding your +1 to the conversation!

**Comment by zachkirsch** [comment]
+1 as well! thanks langfuse team!

**Comment by bajayo** [comment]
+1
But with that, also add a generic "Part" to content based on the provider. 
Example: Google genai "from_bytes" Part:

```
from_bytes_part = types.Part.from_bytes(
      data=base64.b64decode("""JVBERi0xLjQKJ....... (long base64 string)"""),
      mime_type="application/pdf",
  )
```

**Comment by yingzhuivy** [comment]
+1 any update on this?

**Comment by SaudMohammad** [comment]
+1 eagerly waiting on this loving langfuse till now

**Comment by pn-mougel-insa** [comment]
+1 this feature would be great

**Comment by pulkitsharma07** [comment]
+1 on this

**Comment by Maikiii44** [comment]
+1 on this features

**Comment by Arc1el** [comment]
+1 on this

**Comment by eggachecat** [comment]
+1 come on langfuse team!

**Reply by eggachecat** [reply]
I think it should be pretty simple -> compile the prompt "text" with a template engine -> so it support image (something like `<img src=""/>`)

**Comment by matviichukaa** [comment]
+1 on this

**Comment by nickflux** [comment]
+1 from me also
